truth_set/python/structure/solution-approach/semantic-duplication/services/userService.py

```diff
@@ -0,0 +1,19 @@
+from datetime import datetime
+from ..utils.dateUtils import format_date
+
+class UserService:
+    def __init__(self):
+        self.users = []
+    
+    def add_user(self, name):
+        new_user = {
+            'id': len(self.users) + 1,
+            'name': name,
+            'registration_date': datetime.now()
+        }
+        self.users.append(new_user)
+        print(f"User Added: {new_user['name']}, Registered On: {format_date(new_user['registration_date'], 'MM/DD/YYYY')}")
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider extracting common service functionality</b></div>
<div id="fix">

The `add_user` and `add_order` methods appear to have similar functionality for adding items with an ID, timestamp and printing confirmation. Consider creating a common base service class to handle these shared operations. A similar issue was also found in truth_set/python/structure/solution-approach/semantic-duplication/index.py (line 1-17).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
class BaseService:
    def __init__(self):
        self.items = []

    def add_item(self, item_type, data):
        new_item = {
            'id': len(self.items) + 1,
            **data,
            'created_at': datetime.now()
        }
        self.items.append(new_item)
        return new_item

class UserService(BaseService):
    def __init__(self):
        super().__init__()
 
    def add_user(self, name):
        new_user = self.add_item('user', {
            'name': name,
        })
        print(f"User Added: {new_user['name']}, Registered On: {format_date(new_user['registration_date'], 'MM/DD/YYYY')}")
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/5#issuecomment-2553425774>#746476</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged